cpp/src/parquet/encoding.cc

```diff
@@ -2138,8 +2138,6 @@ class DeltaBitPackEncoder : public EncoderImpl, virtual public TypedEncoder<DTyp
           "but it's " +
           std::to_string(values_per_block % mini_blocks_per_block));
     }
-    // Reserve enough space at the beginning of the buffer for largest possible header.
-    PARQUET_THROW_NOT_OK(sink_.Advance(kMaxPageHeaderWriterSize));
```

Comment:
Please make sure it still works correctly if there are zero values (i.e. an empty block).

Reply:
```c++
template <typename DType>
void DeltaBitPackEncoder<DType>::FlushBlock() {
  if (values_current_block_ == 0) {
    return;
  }
```

Don't worry, I will add some tests this weekend...

Reply:
Oh, you're right, I make this wrong. Let me just make it simple

Reply:
I was concerned that there is no sufficient space reserved for the header if the block contains empty values. :)